Lib/email/utils.py

```diff
@@ -106,12 +106,42 @@ def formataddr(pair, charset='utf-8'):
     return address
 
 
+def _pre_parse_validation(fieldvalues):
+    """Validate the field values are syntactically correct"""
+    for v in fieldvalues:
+        s = str(v).replace('\\(', '').replace('\\)', '')
+        if s.count('(') != s.count(')'):
+            fieldvalues.remove(v)
```

Comment:
this loop is modifying the apparent list it is iterating over within the loop.  that makes reasoning about its exact behavior hard.  removing an item could mean you wind up skipping an item, appending an item could make the loop iterate over that.  furthermore remove is O(n)... you're re-finding the item to remove to remove it.  Also, this code modifys the passed in list in place before returning it.

A better code pattern for this is to build up a new list and return that.  never modifying the input.

Reply:
Something like:

```python
accepted_values = []
for v in fieldvalues:
    s = v.replace('\\(', '').replace('\\)', '')
    if s.count('(') != s.count(')'):
        v = ('', '')
    accepted_values.append(v)

return accepted_values
```